### PR TITLE
Card analytics

### PR DESCRIPTION
**Pull request by bsnowden3**

### CODE REVIEW

**Comment by couby1master** [comment]
🔥🔥🔥🔥🔥

**Comment by couby1master** [comment]
🔥🔥🔥🔥🔥🔥😁

**Comment by couby1master** [comment]
🔥🔥🔥🔥🔥🔥🔥🔥🔥😉

File: src/pages/integrations/bitpay-card/bitpay-card-intro/bitpay-card-intro.ts [hunk 1]
```diff
@@ -111,14 +112,17 @@ export class BitPayCardIntroPage {
...
     let url = 'https://bitpay.com/visa/get-started';
     this.externalLinkProvider.open(url);
   }
 
   public connectBitPayCard() {
     if (this.accounts.length == 0) {
+      this.bitPayCardProvider.logEvent('legacycard_connect', {});
       this.startPairBitPayAccount();
     } else {
+      this.bitPayCardProvider.logEvent('legacycard_connect', {});
```
Comment (Gamboster) [inline src/pages/integrations/bitpay-card/bitpay-card-intro/bitpay-card-intro.ts]: If the logEvent is the same as the `if`, maybe we could unify it and put it out of the `if else`

File: src/pages/integrations/bitpay-card/bitpay-card-topup/bitpay-card-topup.ts [hunk 2]
```diff
@@ -512,6 +518,37 @@ export class BitPayCardTopUpPage {
...
+
+  logLegacyCardPurchaseEvent() {
+    this.bitPayCardProvider.logEvent('purchase', {
+      value: this.amount,
+      items: [
+        {
+          name: 'legacyCard',
+          category: 'debitCard',
+          quantity: 1,
+          price: ''
```
Comment (Gamboster) [inline src/pages/integrations/bitpay-card/bitpay-card-topup/bitpay-card-topup.ts]: Is it okay to send the price as an empty string?

File: src/pages/integrations/bitpay-card/bitpay-card-topup/bitpay-card-topup.ts [hunk 3]
```diff
@@ -646,9 +699,18 @@ export class BitPayCardTopUpPage {
     this.remainingTimeStr = ('0' + m).slice(-2) + ':' + ('0' + s).slice(-2);
   }
 
+  logLegacyCardSetCheckoutOption(wallet) {
+    this.bitPayCardProvider.logEvent('set_checkout_option', {
```
Comment (Gamboster) [inline src/pages/integrations/bitpay-card/bitpay-card-topup/bitpay-card-topup.ts]: This log shouldn't be done once the payment is made?
Is it okay to send a new log every time you change the selected wallet?

File: src/providers/bitpay-card/bitpay-card.ts [hunk 4]
```diff
@@ -102,6 +108,10 @@ export class BitPayCardProvider {
     });
   }
 
+  logEvent(eventName: string, eventParams: { [key: string]: any }) {
```
Comment (Gamboster) [inline src/providers/bitpay-card/bitpay-card.ts]: We currently have the `analyticsProvider` ... why don't we use that directly to make the `logEvents`?

**Pull request merged**